exchanges/binance/binance.go

```diff
@@ -321,14 +321,35 @@ func (b *Binance) GetBestPrice(symbol string) (BestPrice, error) {
 // NewOrder sends a new order to Binance
 func (b *Binance) NewOrder(o *NewOrderRequest) (NewOrderResponse, error) {
 	var resp NewOrderResponse
+	if err := b.newOrder(newOrder, o, &resp); err != nil {
+		return resp, err
+	}
+
+	if resp.Code != 0 {
+		return resp, errors.New(resp.Msg)
+	}
 
-	path := b.API.Endpoints.URL + newOrder
+	return resp, nil
+}
+
+// NewOrder sends a new test order to Binance
+func (b *Binance) NewOrderTest(o *NewOrderRequest) error {
+	var resp NewOrderResponse
+	return b.newOrder(newOrderTest, o, &resp)
+}
+
+func (b *Binance) newOrder(api string, o *NewOrderRequest, resp *NewOrderResponse) error {
+	path := b.API.Endpoints.URL + api
 
 	params := url.Values{}
 	params.Set("symbol", o.Symbol)
 	params.Set("side", o.Side)
 	params.Set("type", string(o.TradeType))
-	params.Set("quantity", strconv.FormatFloat(o.Quantity, 'f', -1, 64))
+	if o.QuoteOrderQty > 0 {
+		params.Set("quoteOrderQty", strconv.FormatFloat(o.Quantity, 'f', -1, 64))
```

Comment:
You are not converting the correct field. Should be 
`params.Set("quoteOrderQty", strconv.FormatFloat(o.QuoteOrderQty, 'f', -1, 64))`

Reply:
🤦 must've made this error between stashing/re-editing. Will fix it up

Reply:
Fixed